src/core/SIP/SIPParameters.cs

```diff
@@ -55,7 +56,7 @@ public class SIPParameters
         public char TagDelimiter = DEFAULT_PARAMETER_DELIMITER;
 
         //[DataMember]
-        private Dictionary<string, string> m_dictionary;
+        private ConcurrentDictionary<string, string> m_dictionary;

```

Comment:
Given the other comments, `m_dictionary` can be read-only:

```suggestion
        private readonly ConcurrentDictionary<string, string> m_dictionary;
```